x/ibc/applications/transfer/module.go

```diff
@@ -334,17 +335,27 @@ func (am AppModule) OnRecvPacket(
 		acknowledgement = channeltypes.NewErrorAcknowledgement(err.Error())
 	}
 
-	ctx.EventManager().EmitEvent(
-		sdk.NewEvent(
-			types.EventTypePacket,
-			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
-			sdk.NewAttribute(types.AttributeKeyReceiver, data.Receiver),
-			sdk.NewAttribute(types.AttributeKeyDenom, data.Denom),
-			sdk.NewAttribute(types.AttributeKeyAmount, fmt.Sprintf("%d", data.Amount)),
-			sdk.NewAttribute(types.AttributeKeyAckSuccess, fmt.Sprintf("%t", err != nil)),
-		),
+	channel, _ := am.keeper.GetChannel(ctx, packet.GetSourcePort(), packet.GetSourceChannel())
+
+	err = ctx.EventManager().EmitTypedEvent(
```

Comment:
this should be emitted in the channel keeper. See `x/ibc/core/04-channel/keeper/packet.go#RecvPacket`. client/connection/channel is core IBC stuff and we want to always emit that info in core IBC so application developers don't have to duplicate this code